### DISCUSSION TITLE
Cache Flush on Failure Indication

### DISCUSSION THREAD

**Discussion by keepsimple1** [keepsimple1/mdns-sd#116] [Polls]
Triggered by issue #54 , I'm looking into how we should support ["Cache Flush on Failure Indication"](https://datatracker.ietf.org/doc/html/rfc6762#section-10.4): i.e. a faster way to remove stale data from the daemon cache.

I'm wondering which option would fit most users' needs better. Hopefully this poll can help ;-) (This is my first time using a poll on GitHub).

**Comment by dalepsmith** [comment]
The Apple Bonjour API has `DNSServiceReconfirmRecord`.
I don't think Avahi has anything like that.

The Apple call seems very low level, you need to pass in raw rodata.

I found this, but I'm not sure how up-to-date it is: https://developer.apple.com/documentation/dnssd/1804726-dnsservicereconfirmrecord?language=objc

**Reply by keepsimple1** [reply]
Thanks for the pointer! Indeed the Apple's API seems quite low level with many parameters. It seems to be up-to-date, as it has a Swift version without any deprecate messages.

**Comment by dalepsmith** [comment]
Has anyone ever seen this actually used in the wild?

What record(s) should this be used with?  Ok, looking at the RFC:

- The A record if arp requests fail, host unreachable, etc.
- The SRV record if can't communicate with the host on the port.
- The PTR record if looking up the associated SRV fails.

**Reply by keepsimple1** [reply]
I don't know if people actually re-confirm the individual records. I think it's more practical to verify/re-confirm a Service Instance, automatically including its SRV, A, and TXT records. If it cannot be resolved anymore, then its PTR record should also be removed.  From the API point of view, the input would be the full name of the Service Instance.

> Has anyone ever seen this actually used in the wild?

This is a good question. We probably should not implement this until see actual use cases.
